cmd/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/arraial/pipo-dispatcher/internal/settings"
+	"github.com/arraial/pipo-dispatcher/internal/telemetry"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+var version string = "latest"
+
+func main() {
+	var config = settings.InitConfig()
+
+	fmt.Println(config.GetString("telemetry.service"))
+	fmt.Println("Version: " + version)
+
+	if err := run(); err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func serverIsHealthy() bool {
+	return true
+}
+
+func healthHandler(w http.ResponseWriter, r *http.Request) {
+	if serverIsHealthy() {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, "Server is healthy")
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprint(w, "Server is not healthy")
+	}
+}
+
+func newHTTPHandler() http.Handler {
+	mux := http.NewServeMux()
+
+	handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
+		handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+		mux.Handle(pattern, handler)
+	}
+
+	handleFunc("/healthz", healthHandler)
+
+	handler := otelhttp.NewHandler(mux, "/")
+	return handler
+}
+
+func run() (err error) {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	otelShutdown, err := telemetry.SetupOTelSDK(ctx)
+	if err != nil {
+		return
+	}
+	defer func() {
+		err = errors.Join(err, otelShutdown(context.Background()))
+	}()
+
+	// Start HTTP server.
+	srv := &http.Server{
+		Addr:         ":8080",
+		BaseContext:  func(_ net.Listener) context.Context { return ctx },
+		ReadTimeout:  time.Second,
+		WriteTimeout: 10 * time.Second,
+		Handler:      newHTTPHandler(),
+	}
+	srvErr := make(chan error, 1)
+	go func() {
+		srvErr <- srv.ListenAndServe()
+	}()
+
+	// Handle interruptions
+	select {
+	case err = <-srvErr:
+		return
+	case <-ctx.Done():
+		stop()
```

Comment:
Log something here, otherwise it'll be hard to understand when a container died abruptly or when it was gracefully killed